python/activator/middleware_interface.py

```diff
@@ -638,11 +638,15 @@ def _export_calibs(self, detector_id, filter):
         """
         # TAI observation start time should be used for calib validity range.
         calib_date = astropy.time.Time(self.visit.private_sndStamp, format="unix_tai")
+        # Querying by specific types is much faster than querying by ...
+        # Some calibs have an exposure ID (of the source dataset?), but these can't be used in AP.
```

Comment:
FWIW, calibs with an `exposure` dimension don't make sense to me either.  I wonder if they're accidents.

Reply:
It seems to be a mix of things:

- Produced by `IsrTask` or `CalibCombineTask` in CPP:
    - cpBiasProc, biasIsr_metadata, biasIsr_log
    - cpDarkIsr, darkIsr_metadata, darkIsr_log
    - cpDarkProc, cpDark_metadata, cpDark_log
    - cpDarkProcForDefects, darkIsrForDefects_metadata, darkIsrForDefects_log
    - cpFlatProc, flatIsr_metadata, flatIsr_log
    - flatStats, cpFlatMeasure_metadata, cpFlatMeasure_log
    - cpDefectsProc
    - cpPartialDefects, measureDefects_metadata, measureDefects_log
    - cpPtcProc, ptcIsr_metadata, ptcIsr_log
    - cpPtcExtract
    - cpCrosstalkProc
    - crosstalkRatios, crosstalkFluxes, cpCrosstalkRatio, crosstalkExtract_metadata, crosstalkExtract_log
- Produced in the verification counterparts of the above:
    - verifyBiasProc, verifyBiasApply_metadata, verifyBiasApply_log
    - verifyBiasDetStats, verifyMeasureStats_metadata, verifyMeasureStats_log
    - verifyBiasExpStats, verifyExposureStats_log, verifyExposureStats_metadata
    - verifyDarkProc, verifyDarkApply_metadata, verifyDarkApply_log
    - verifyDarkDetStats, verifyDarkChip_metadata, verifyDarkChip_log
    - verifyDarkExpStats, verifyDarkExp_metadata, verifyDarkExp_log
    - verifyFlatProc, verifyFlatApply_metadata, verifyFlatApply_log
    - verifyFlatDetStats, verifyFlatChip_metadata, verifyFlatChip_log
    - verifyFlatExpStats, verifyFlatExp_metadata, verifyFlatExp_log
    - verifyDefectProc, verifyDefectApply_metadata, verifyDefectApply_log
    - verifyDefectDetStats
    - verifyDefectExpStats
    - verifyDefectIsr, verifyDefectIsr_metadata, verifyDefectIsr_log
    - verifyUncorrDefectIsr, verifyUncorrectedDefectIsr_metadata, verifyUncorrectedDefectIsr_log
    - verifyCrosstalkProc, verifyCrosstalkApply_metadata, verifyCrosstalkApply_log
    - verifyCrosstalkRatio, verifyCrosstalkExtract_metadata, verifyCrosstalkExtract_log
    - verifyLinearityProc, verifyLinearityApply_metadata, verifyLinearityApply_log
    - verifyLinearityPtcExtract
- Miscellaneous:
    - raw
    - rawSpectrum
    - postISRCCD, isrStatistics, isr_metadata, isr_log
    - timing_isr_metadata, timing_isr_log, cputiming_isr_metadata, cputiming_isr_log [Did somebody run `ApVerify.yaml` on /repo/embargo???]
- Unidentified (possibly obsolete):
    - quickLookExp, quickLookTask_metadata, quickLookTask_log
    - crosstalkSignals, crosstalkResults, crosstalkBackgroundXTilts, crosstalkBackgroundYTilts, crosstalkBackgroundZOffsets, crosstalkCoefficients, crosstalkCoefficientErrors, crosstalkNonLinearCoefficients, crosstalkTask_metadata, crosstalkTask_log
    - backgroundResults
    - plotPtcTask_metadata, plotPtcTask_log
    - cpCRisr, cpCRproc
    - spotSrc, characterizeSpots_metadata, characterizeSpots_log
    - eoFlatIsr, eoFlatStats, eoFlatMeasure_metadata, eoFlatMeasure_log
    - eoDarkIsr, eoDarkProc, eoDarkTask_log, eoDarkTask_metadata
    - myIsr_metadata, myIsr_log
    - verifyBiasExpMatrix, verifyBiasExpResults, verifyBiasExp_metadata, verifyBiasExp_log
    - verifyBiasDetMatrix, verifyBiasDetResults, verifyBiasChip_metadata, verifyBiasChip_log
    - verifyDarkExpResults, verifyDarkDetResults
    - verifyFlatDetResults, verifyFlatExpResults

Reply:
I would expect bias/dark/flat calibrations to all be exposure calibrations and not visit.